library/src/main/java/org/apache/apex/malhar/lib/state/managed/IncrementalCheckpointManager.java

```diff
@@ -109,8 +110,8 @@ public void run()
           transferWindowFiles();
           if (latestExpiredTimeBucket.get() > -1) {
             try {
-              managedStateContext.getBucketsFileSystem().deleteTimeBucketsLessThanEqualTo(
-                  latestExpiredTimeBucket.getAndSet(-1));
+              latestPurgedTimeBucket = latestExpiredTimeBucket.getAndSet(-1);
```

Comment:
In the case of windowed operator, this is for for event-time windows that are beyond the lateness horizon at the time when the streaming window is committed. In that case, it's not possible that the data deleted here may be needed in the future. This may not be true for other usages, and maybe that's why the unit test is failing. I need suggestion on how to do this the correct way.